Content.Server/GameObjects/Components/Janitorial/LightReplacerComponent.cs

```diff
@@ -0,0 +1,188 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Content.Server.GameObjects.Components.Janitorial
+{
+    /// <summary>
+    ///     Device that allows user to quikly change bulbs in <see cref="PoweredLightComponent"/>
+    ///     Can be reloaded by new light tubes or light bulbs
+    /// </summary>
+    [RegisterComponent]
+    public class LightReplacerComponent : Component, IAfterInteract, IInteractUsing
+    {
+        public override string Name => "LightReplacer";
+
+        [DataField("contents")] private List<LightReplacerEntity> _contents = new();
+        [DataField("sound")] private string _sound = "/Audio/Weapons/click.ogg";
+
+        [ViewVariables] private IContainer _bulbsStorage = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            // fill light replacer container with bulbs
+            _bulbsStorage = ContainerHelpers.EnsureContainer<Container>(Owner, "light_replacer_storage");
+            foreach (var ent in _contents)
+            {
+                for (var i = 0; i < ent.Amount; i++)
+                {
+                    var bulb = Owner.EntityManager.SpawnEntity(ent.PrototypeName, Owner.Transform.Coordinates);
+                    _bulbsStorage.Insert(bulb);
+                }
+            }
+        }
+
+        async Task<bool> IAfterInteract.AfterInteract(AfterInteractEventArgs eventArgs)
+        {
+            // standard interaction checks
+            if (!ActionBlockerSystem.CanUse(eventArgs.User)) return false;
+            if (!eventArgs.CanReach) return false;
+
+            // behaviour will depends on target type
+            if (eventArgs.Target != null)
+            {
+                // replace broken light in fixture?
+                if (eventArgs.Target.TryGetComponent(out PoweredLightComponent? fixture))
+                    return TryReplaceBulb(fixture, eventArgs.User);
+                // add new bulb to light replacer container?
+                else if (eventArgs.Target.TryGetComponent(out LightBulbComponent? bulb))
+                    return TryInsertBulb(bulb, eventArgs.User, true);
+            }
+
+            return false;
+        }
+
+        async Task<bool> IInteractUsing.InteractUsing(InteractUsingEventArgs eventArgs)
+        {
+            // standard interaction checks
+            if (!ActionBlockerSystem.CanInteract(eventArgs.User)) return false;
+
+            if (eventArgs.Using != null)
+            {
+                // want to insert a new light bulb?
+                if (eventArgs.Using.TryGetComponent(out LightBulbComponent? bulb))
+                    return TryInsertBulb(bulb, eventArgs.User, true);
+                // add bulbs from storage?
+                else if (eventArgs.Using.TryGetComponent(out ServerStorageComponent? storage))
+                    return TryInsertBulb(storage, eventArgs.User);
+            }
+
+            return false;
+        }
```

Comment:
If you can put these into a LightReplacerSystem EntitySystem using the SubscribeLocalEvent. Like how BuckleSystem does it:
https://github.com/space-wizards/space-station-14/blob/f3a180c3d5d3f9c503fd7b61815611fc42e4eda6/Content.Server/GameObjects/EntitySystems/BuckleSystem.cs#L27